sdk/storage/blobs/samples/src/blobs_client_example.c

```diff
@@ -52,7 +62,15 @@ az_storage_blobs_blob_download(az_storage_blobs_blob_client* client, az_http_res
       AZ_SPAN_NULL));
 
   // start pipeline
-  return az_http_pipeline_process(&client->_internal.pipeline, &hrb, response);
+  AZ_RETURN_IF_FAILED(az_http_pipeline_process(&client->_internal.pipeline, &hrb, response));
```

Comment:
What is this? Customers should NOT have to know about the pipieline or use _internal fields.

Reply:
This isn't really customer code.  This is the download method which would be part of blob client.  The client methods internally kick off the pipelines by calling this method and passing the start of the pipeline.